build-telegraf/plugins.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+# Determine the root project directory
+ROOT_DIR=$(git rev-parse --show-toplevel)
+
+# Clone an external repository
+git clone https://github.com/influxdata/telegraf.git "$ROOT_DIR/external/telegraf"
+
+# Update the external repository
+cd "$ROOT_DIR/external/telegraf"
+git pull origin master
+
+# Function to create a Hugo page
+create_page() {
+  local plugin_dir=$1
+  local dest_dir=$2
+  local dest_file="$dest_dir/_index.md"
+  local parent=$3
+
+  # Check if the plugin directory exists
+  if [ -d "$plugin_dir" ]; then
+    mkdir -p "$dest_dir"
+    # Use a heredoc to write the frontmatter to the destination file
+    cat <<EOF > "$dest_file"
+---
+description: "Telegraf plugin for collecting metrics from $plugin_dir"
+menu:
+  telegraf_v1_ref:
+    parent: $parent
+    name: $(basename "$plugin_dir")
+tags: [$(basename "$plugin_dir")]
+---
+
+EOF
+  else
+    echo "Plugin directory not found: $plugin_dir"
```

Comment:
We currently have 20 external plugins that are listed on the plugins page but are not part of the Telegraf repo. How do we want to handle these? @srebhan, do you have any thoughts?